trunk/SUAVE/Components/Energy/Storages/Batteries/Constant_Mass/Lithium_Ion.py

```diff
@@ -3,31 +3,351 @@
+                voltage_under_load                    [volts]
+            state.unknowns.battery_voltage_under_load [volts]
+            
+            Outputs:
+            None
+    
+            Properties Used:
+            network.voltage                           [volts]
+        """     
+        v_actual  = segment.state.conditions.propulsion.battery_voltage_under_load
+        v_predict = segment.state.unknowns.battery_voltage_under_load
+        v_max     = network.voltage
+        
+        # Return the residuals
+        segment.state.residuals.network.voltage = (v_predict[:,0] - v_actual[:,0])/v_max
+        
+        return 
+    
+    def append_battery_unknowns_and_residuals_to_segment(self,segment,initial_voltage, 
+                                              initial_battery_cell_temperature , initial_battery_state_of_charge,
+                                              initial_battery_cell_current,initial_battery_cell_thevenin_voltage): 
+        """ Sets up the information that the mission needs to run a mission segment using this network
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:  
+            initial_voltage                       [volts]
+            initial_battery_cell_temperature      [Kelvin]
+            initial_battery_state_of_charge       [unitless]
+            initial_battery_cell_current          [Amperes]
+            initial_battery_cell_thevenin_voltage [Volts]
+            
+            Outputs
+            None
+            
+            Properties Used:
+            N/A
+        """        
+        
+        ones_row = segment.state.ones_row 
+        if initial_voltage==None:
+            initial_voltage = self.max_voltage 
+        segment.state.unknowns.battery_voltage_under_load  = initial_voltage * ones_row(1) 
+        
+        return  
+    
+    def compute_voltage(self,state):
+        """ Computes the voltage of a single LFP cell or a battery pack of LFP cells   
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:  
+                self    - battery data structure             [unitless]
+                state   - segment unknowns to define voltage [unitless]
+            
+            Outputs
+                V_ul    - under-load voltage                 [volts]
+             
+            Properties Used:
+            N/A
+        """              
+        # Unpack battery properties
+        battery                          = self 
+        
+        # Set battery properties
+        battery.battery_thevenin_voltage = 0     
+        battery.temperature              = state.conditions.propulsion.battery_pack_temperature 
+        
+        # Voltage under load
+        V_ul                             = state.unknowns.battery_voltage_under_load
+        return V_ul  
+    
+    def update_battery_cycle_day(self,segment):    
```

Comment:
should there be a warning here that this function isn't doing anything?

Reply:
corrected